tests/spread/general/sources/snaps/git-submodules/snapcraft.yaml

```diff
@@ -0,0 +1,15 @@
+name: git-recurse-submodules
+base: core20
+version: "0.1"
+summary: Test the use of source-submodules
+description: Make sure source-submodules works
+confinement: strict
+
+parts:
+    git:
+        plugin: dump
+        source: https://github.com/snapcore/core18
+        source-type: git
+        source-submodules:
+          - submodule_1
```

Comment:
these submodules exist?

Reply:
No.  Using `--recurse-submodules=<non-existant-directory>` with `git clone` doesn't produce any errors.  The lack of error or warning surprises me, but from what I gather, it's intentional. 

This test only proves that the parameters are parsed and the `git clone` command runs.

Perhaps a slightly better test would be to choose a different repo with a submodule.  Unfortunately I don't see any in snapcore.  The multipass repo has some submodules.  Alternatively I could clone a repo from an outside project.  Think it's worth the trouble?